target/ppc/excp_helper.c

```diff
@@ -2172,6 +2172,92 @@ void helper_td(CPUPPCState *env, target_ulong arg1, target_ulong arg2,
 #endif
 #endif
 
+static uint32_t helper_SIMON_LIKE_32_64(uint32_t x, uint64_t key, uint32_t lane)
+{
+    uint32_t c = 0xfffc;
+    uint64_t z0 = 0xfa2561cdf44ac398;
+    uint16_t z = 0, temp;
+    uint16_t k[32], eff_k[32], xleft[33], xright[33], fxleft[32];
+
+    for (int i = 3; i >= 0; i--) {
+        k[i] = key & 0xffff;
+        key >>= 16;
+    }
+    xleft[0] = x & 0xffff;
+    xright[0] = (x >> 16) & 0xffff;
+
+    for (int i = 0; i < 28; i++) {
+        z |= ((z0 >> (63 - i)) & 1) << 48;
```

Comment:
`<< 48` was because it's setting 15th bit and I assumed it was a 64 bit variable, but it's uint16. Fixed by removing it for now, but I have yet to take a look into the bit ordering issue